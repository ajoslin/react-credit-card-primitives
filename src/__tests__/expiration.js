const React = require('react')
const {mount} = require('enzyme')
const Expiration = require('../expiration')

function setup ({render = () => <div />, ...props} = {}) {
  let renderArg
  const renderSpy = jest.fn(arg => {
    renderArg = arg
    return render(arg)
  })
  const wrapper = mount(<Expiration {...props} render={renderSpy} />)
  return {renderSpy, wrapper, ...renderArg}
}

test('basic value', () => {
  const {wrapper, setRawValue, getInputProps} = setup()

  setRawValue('10/2022')
  expect(getInputProps().value).toEqual('10 / 22')
  expect(getInputProps()).toMatchSnapshot()
})
