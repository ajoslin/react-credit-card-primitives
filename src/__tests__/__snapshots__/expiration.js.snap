// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic value 1`] = `
Object {
  "aria-invalid": "false",
  "autoComplete": "exp-date",
  "maxLength": 9,
  "name": "cc-exp",
  "onChange": [Function],
  "placeholder": "MM / YY",
  "type": "text",
  "value": "10 / 22",
}
`;
